dbt/context/parser.py

```diff
@@ -50,11 +50,18 @@ def __init__(self, model, source_config):
         self.model = model
         self.source_config = source_config
 
+    @staticmethod
+    def _transform_kwargs(kwargs):
+        for k in ('pre_hook', 'post_hook'):
+            if k in kwargs:
+                kwargs[k.replace('_', '-')] = kwargs.pop(k)
+        return kwargs
```

Comment:
@drewbanin is there a concern about `post_hook` overwriting `post-hook` if both are specified

Reply:
It will only do so if you pass it as `kwargs`, not a dict. You have to really work to get into that situation (set a dict with both keys as a variable and `**`-expand it, I think).

Reply:
My bigger concern would be that `post_hook` keys are ignored when `config()` is called with a dictionary. I think that's ok... but it would be nice to at least warn, since that behavior is confusing. Same for `pre_hook`

Reply:
I can also just move the call down a line and operate on `opts` instead of `kwargs` to support both, but then we should probably error on getting both instead of just overwriting.

Reply:
yeah, that behavior sounds ideal @beckjake